lib/plugin/instance.rb

```diff
@@ -797,15 +797,33 @@ def add_api_key_scope(resource, action)
   # in a query parameter rather than a header. For example:
   #
   # add_api_parameter_route(
-  #   method: :get,
-  #   route: "users#bookmarks",
-  #   format: :ics
+  #   methods: :get,
+  #   actions: "users#bookmarks",
+  #   formats: :ics
   # )
   #
   # See Auth::DefaultCurrentUserProvider::PARAMETER_API_PATTERNS for more examples
   # and Auth::DefaultCurrentUserProvider#api_parameter_allowed? for implementation
-  def add_api_parameter_route(method:, route:, format:)
-    DiscoursePluginRegistry.register_api_parameter_route({ method: method, route: route, format: format }, self)
+  def add_api_parameter_route(method: nil, methods: nil,
+                              route: nil, actions: nil,
+                              format: nil, formats: nil)
+
+    if Array(format).include?("*")
+      # Old API used * as wildcard. New api uses `nil`
+      format = nil
+    end
+
+    # Backwards compatibility with old parameter names:
+    methods ||= method
+    actions ||= route
+    formats ||= format
+
+    DiscoursePluginRegistry.register_api_parameter_route(
+      RouteMatcher.new(
+        methods: methods,
+        actions: actions,
+        formats: formats
+      ), self)
```

Comment:
Do we want to drop the old parameters at some point? If so, maybe we should a deprecation notice?